API.md

```diff
@@ -0,0 +1,269 @@
+# API
+
+* [`create`](#create)
+* [`start`](#start)
+* [`stop`](#stop)
+* [`dial`](#dial)
+* [`dialProtocol`](#dialProtocol)
+* [`hangUp`](#hangUp)
+* [`handle`](#handle)
+* [`unhandle`](#unhandle)
+* [`peerRouting.findPeer`](#peerRouting.findPeer)
+* [`contentRouting.findProviders`](#contentRouting.findProviders)
+* [`contentRouting.provide`](#contentRouting.provide)
+* [`dht.put`](#dht.put)
+* [`dht.get`](#dht.get)
+* [`dht.findPeer`](#dht.findPeer)
+* [`dht.provide`](#dht.provide)
+* [`dht.findProviders`](#dht.findProviders)
+* [`dht.getClosestPeers`](#dht.getClosestPeers)
+* [`dht.getPublicKey`](#dht.getPublicKey)
+* [`pubsub.getPeersSubscribed`](#pubsub.getPeersSubscribed)
+* [`pubsub.getTopics`](#pubsub.getTopics)
+* [`pubsub.publish`](#pubsub.publish)
+* [`pubsub.subscribe`](#pubsub.subscribe)
+
+## create
+
+Creates an instance of Libp2p.
+
+### `create(options)`
+
+#### Parameters
+
+| Name | Type | Description |
+|------|------|-------------|
+| options | `Object` | libp2p options |
+
+#### Returns
+
+| Type | Description |
+|------|-------------|
+| `Promise<Libp2p>` | Promise resolves with the Libp2p instance |
+
+#### Example
+
+```js
+const { create } = require('libp2p')
+
+// specify options
+const options = {}
+
+// create libp2p
+const libp2p = await create(options)
+```
+
+Note: It is important pointing out that with `create`, the `PeerInfo` option is not required and will be generated if it is not provided.
+
+As an alternative, it is possible to create a Libp2p instance with the constructor:
+
+#### Example
+
+```js
+const Libp2p = require('libp2p')
+
+// specify options
+const options = {}
+
+// create libp2p
+const libp2p = new Libp2p (options)
+```
+
+Required keys in the `options` object:
+
+- `peerInfo`: instance of [PeerInfo][] that contains the [PeerId][], Keys and [multiaddrs][multiaddr] of the libp2p Node.
+- `modules.transport`: An array that must include at least 1 transport, such as `libp2p-tcp`.
+
+## start
+
+Starts the libp2p node.
+
+### `libp2p.start()`
+
+#### Returns
+
+| Type | Description |
+|------|-------------|
+| `Promise` | Promise resolves when the node is ready |
+
+#### Example
+
+```js
+const { create } = require('libp2p')
+
+// ...
+
+const libp2p = await create(options)
+
+// start libp2p
+await libp2p.start()
+```
+
+## stop
+
+Stops the libp2p node.
+
+### `libp2p.stop()`
+
+#### Returns
+
+| Type | Description |
+|------|-------------|
+| `Promise` | Promise resolves when the node is fully stopped |
+
+#### Example
+
+```js
+const { create } = require('libp2p')
+
+// ...
+const libp2p = await create(options)
```

Comment:
A matter of preference but IMHO calling create on a default export is clearer in examples (and in app code):

```suggestion
const libp2p = await Libp2p.create(options)
```